tests/metrics/test_mortality_observer.py

```diff
@@ -77,91 +83,75 @@ def simulation_after_one_step(base_config, base_plugins):
-    pop = simulation_after_one_step.get_population()
+    """Test that all expected observation stratifications appear in the results."""
+    simulation_after_one_step.finalize()
+    simulation_after_one_step.report()
+    results_dir = Path(simulation_after_one_step.configuration.output_data.results_directory)
+    results_files = list(results_dir.rglob("*.parquet"))
+    assert set(file.name for file in results_files) == set(["deaths.parquet", "ylls.parquet"])
+    deaths = pd.read_parquet(results_dir / "deaths.parquet")
+    ylls = pd.read_parquet(results_dir / "ylls.parquet")
+
+    expected_stratifications = set(
+        itertools.product(*[["other_causes", "flu", "mumps"], ["Female", "Male"]])
+    )
+    assert set(zip(deaths[COLUMNS.ENTITY], deaths["sex"])) == expected_stratifications
+    assert set(zip(ylls[COLUMNS.ENTITY], ylls["sex"])) == expected_stratifications
+
 
-    expected_observations = [
-        "MEASURE_death_due_to_other_causes_SEX_Female",
-        "MEASURE_death_due_to_other_causes_SEX_Male",
-        "MEASURE_ylls_due_to_other_causes_SEX_Female",
-        "MEASURE_ylls_due_to_other_causes_SEX_Male",
-        "MEASURE_death_due_to_flu_SEX_Female",
-        "MEASURE_death_due_to_flu_SEX_Male",
-        "MEASURE_ylls_due_to_flu_SEX_Female",
-        "MEASURE_ylls_due_to_flu_SEX_Male",
-        "MEASURE_death_due_to_mumps_SEX_Female",
-        "MEASURE_death_due_to_mumps_SEX_Male",
-        "MEASURE_ylls_due_to_mumps_SEX_Female",
-        "MEASURE_ylls_due_to_mumps_SEX_Male",
-    ]
-
-    assert set(expected_observations) == set(results(pop.index).keys())
-
-
-@pytest.mark.skip(reason="MIC-4981: update components for new results processing")
 def test_observation_correctness(simulation_after_one_step):
     """Test that deaths and YLLs appear as expected in the results."""
-    expected_deaths, expected_ylls = get_expected_results(simulation_after_one_step)
-    results = simulation_after_one_step.get_value("metrics")
-    pop = simulation_after_one_step.get_population()
-
-    for observation in expected_deaths:
-        assert np.isclose(
-            results(pop.index)[observation], expected_deaths[observation], rtol=0.001
-        )
-    for observation in expected_ylls:
-        assert np.isclose(
-            results(pop.index)[observation], expected_ylls[observation], rtol=0.001
-        )
+    expected = get_expected_results(simulation_after_one_step)
+    _assert_metric_correctness(simulation_after_one_step, expected)
 
     # same test on second time step
     simulation_after_one_step.step()
-    expected_deaths, expected_ylls = get_expected_results(
-        simulation_after_one_step,
-        expected_deaths=expected_deaths,
-        expected_ylls=expected_ylls,
+    expected = get_expected_results(simulation_after_one_step, expected)
+    _assert_metric_correctness(simulation_after_one_step, expected)
+
+    # ensure actual output files match as well
```

Comment:
This is a little silly. We typically have been testing correctness against the output/written datasets but I didn't want to refactor everything here so I kept w/ comparing it to the metrics pipeline. But then I did want to compare the written results as well so here we are